.github/workflows/test.yml

```diff
@@ -18,10 +22,31 @@ jobs:
     - uses: actions/checkout@v2
 
     - name: Setup Node.js environment
-      uses: actions/setup-node@v1.4.2
+      uses: actions/setup-node@v1
+      with:
+        node-version: 15.x
 
-    - name: Install dependencies
+    - name: Cache ~/.elm
+      uses: actions/cache@v2
+      with:
+        path: ~/.elm
+        key: elm-ubuntu-${{ hashFiles('**/elm.json', 'elm-tooling.json') }}
+
+    - name: Cache node_modules
+      id: cache-node_modules
+      uses: actions/cache@v2
+      with:
+        path: node_modules
+        key: node_modules-ubuntu-${{ hashFiles('package-lock.json') }}
+
+    - name: npm ci
+      if: steps.cache-node_modules.outputs.cache-hit != 'true'
+      env:
+        NO_ELM_TOOLING_INSTALL: 1
```

Comment:
What happens if `elm-tooling.json` is updated? For instance to say that we need `elm-format` v0.9.0. Shouldn't `elm-tooling` re-install the tools?

Reply:
The tools listed in `elm-tooling.json` are installed in the next step. I’ve tried to explain CI setup here: https://github.com/lydell/elm-tooling.json/tree/main/cli#ci

Reply:
Ah, so setting `NO_ELM_TOOLING_INSTALL` doesn't prevent the next step.
Should the result of `elm-tooling install` not be cached?

Reply:
> Ah, so setting NO_ELM_TOOLING_INSTALL doesn't prevent the next step.

Correct, were setting env vars per step here, and they don’t propagate to the next steps.

> Should the result of `elm-tooling install` not be cached?

`elm-tooling install` downloads stuff to `~/.elm/elm-tooling/` and we cache `~/.elm` above.

`elm-tooling install` also generates “links” to the downloaded binaries in `./node_modules/.bin/`, but that’s so fast that trying to avoid that work through caching doesn’t bring any time savings.

Reply:
> elm-tooling install downloads stuff to ~/.elm/elm-tooling/ and we cache ~/.elm above

Alright, I thought that the action cached the contents of `~/.elm` as it was at the moment that step was run. So since `elm-tooling` downloads the binaries later on, I thought we cached an empty folder.

But it caches it at the end of the run but restores the cache at this step, is that right?

Reply:
That is correct! You can see that it works that way here: https://github.com/jfmengels/node-elm-review/pull/28/checks?check_run_id=1436496301